src/subgraph/fully-connected.c

```diff
@@ -322,6 +322,62 @@ static enum xnn_status create_fully_connected_operator(
   return status;
 }
 
+static enum xnn_status resize_fully_connected_output_tensor(
+  const struct xnn_operator_data* opdata,
+  struct xnn_value* values,
+  size_t num_values,
+  size_t old_workspace_size,
+  pthreadpool_t threadpool)
+{
+  const uint32_t filter_id = opdata->inputs[1];
+  const struct xnn_value* filter = &values[filter_id];
+
+  const uint32_t output_id = opdata->outputs[0];
+  struct xnn_value* output = (struct xnn_value*) &values[output_id];
+
+  const uint32_t input_id = opdata->inputs[0];
+  const struct xnn_value* input = &values[input_id];
+
+  // Infer output channels.
+  const uint32_t output_channel_index = (opdata->flags & XNN_FLAG_TRANSPOSE_WEIGHTS) ? 1 : 0;
+  enum xnn_shape_inference_status status =
+    xnn_tensor_propagate_dimension(output, output->shape.num_dims - 1, filter->shape.dim[output_channel_index]);
```

Comment:
Make this /*to_dim=*/ 1 and move the assert from line 352 before this

Reply:
Not sure I follow, IIUC output rank can be >2 here.

Reply:
True, output can have > 2 rank